### ISSUE TITLE
Cross-folder type resolution

### ISSUE THREAD

**Issue by bpunsky** [gingerBill/Odin#83]
This one perplexed me for a bit. I finally figured out that it was because I was referencing files from different folders, and that was somehow messing with type resolution. The example below works as expected; it prints "TestTestTest!". If you uncomment the import and comment the one above it, you get:

> E:\Workspace\test\test.odin(7:19) Cannot assign value `&f` of type `^Foo` to `^Foo` in argument

despite there being no difference between the two bar.odin files, except for how they reference foo.odin and which folder they are in. Seemingly a problem with imports being circular with regard to the folder?

folder structure:
```
test/
    folder/
        bar.odin
    bar.odin
    foo.odin
    test.odin
```

test/folder/bar.odin:
```
import "../foo.odin";
import "fmt.odin";

print_foo :: proc(f: ^foo.Foo) {
    fmt.println(f.str);
}
```

test/bar.odin:
```
import "foo.odin";
import "fmt.odin";

print_foo :: proc(f: ^foo.Foo) {
    fmt.println(f.str);
}
```

test/foo.odin:
```
Foo :: struct {
    str: string;
}
```

test/test.odin:
```
import "foo.odin";
import "bar.odin";
//import "folder/bar.odin";

main :: proc() {
    f := foo.Foo{"TestTestTest!"};

    bar.print_foo(&f);
}
```

**Comment by gingerBill** [comment]
A better error message here would be a good idea.

**Comment by bpunsky** [comment]
Is it intended that you can't use imports this way, then?

**Comment by bpunsky** [comment]
Confirmed that it's not circularity that's the problem. In the example, if you move foo.odin up a directory and change the imports accordingly, the same error occurs. Seems to be caused by referencing the same file from different folders; perhaps because the import string is different, it's not seen as the same file, and thus the types are not equivalent?

**Comment by gingerBill** [comment]
Maybe. Which operating system is this? Linux? It might be the problem that it file gets checked twice and thus, the entities are not the same because the compiler thinks it is a different file.  However, I personally do not officially support *nix yet. I will do eventually when I have the Windows version working very well.

**Comment by bpunsky** [comment]
No, it's Windows 10. It's not a breaking issue at this point because everything works fine as long as it's in the same folder, but obviously it would be nice to be able to have a reasonable folder structure for organization purposes.

Maybe this would be better illustrated by a real use case - I have a file game.odin that has a GameComponent struct. I have three files which define components, which use the Game and GameComponent structs from game.odin. And then I have a main file that adds the components to the game and starts the game loop. I tried to put the components in their own folder, and that's how I found this (bug?). The game.odin file was imported from main.odin its own folder and the components in the components/ folder, so I got 

> Cannot assign value &game of type ^Game to ^Game in argument

from main.odin for each of the components. Having the components in the same folder as the main engine files seems a little bit messy to me, but I can see how a circular folder structure could be undesirable as well. So if this is the intended functionality, that's fine, but the ^Game != ^Game error message leads me to believe this is a bug.
